models/result_value.go

```diff
@@ -21,3 +21,13 @@ func (r *ResultValue) GetScalarType() string {
 		return "other"
 	}
 }
+
+type ResultValueWrap struct {
+	EntryWrap *Entry
+	Value     *ResultValue
+}
+
+func (e *Entry) WrapResultValue(val ResultValue) ResultValueWrap {
+	var rvWrap = ResultValueWrap{EntryWrap: e, Value: &val}
+	return rvWrap
```

Comment:
I can do that first bit.

With the second point though, I worry about circular dependencies. I'll try it when I'm back at my computer though.